frame/staking/src/offchain_election.rs

```diff
@@ -155,19 +166,123 @@ pub fn get_balancing_iters<T: Trait>() -> usize {
 	}
 }
 
+/// Find the maximum `len` that a compact can have in order to fit into the block weight.
+pub fn maximum_compact_len<T: Trait>(
+	winners_len: u32,
+	size: ElectionSize,
+	max_weight: Weight,
+) -> u32 {
+	// TODO: ofc we can do this nicer, if we rely on the internals of the weight function :\
+	// TODO: check infinite loop is not possible.
+	let mut voters = 0;
+	loop {
+		let weight = T::WeightInfo::submit_solution_better(
+			size.validators.into(),
+			size.nominators.into(),
+			voters,
+			winners_len,
+		);
+		if weight < max_weight {
+			voters += 1;
```

Comment:
Feels like we should be doing a binary search

Reply:
Isnt there a known upper limit for this value? This loop assumes no upper limit

Reply:
Yeah I have a TODO: comment on top of it regarding both concerns.

Reply:
Added it, it looks a bit strange to me though... but I tested it thoroughly.